domain/src/main/java/org/oppia/android/domain/exploration/lightweightcheckpointing/ExplorationCheckpointController.kt

```diff
@@ -69,69 +59,79 @@ class ExplorationCheckpointController @Inject constructor(
   /**
    * Records an exploration checkpoint for the specified profile.
    *
-   * @return a [DataProvider] that indicates the success/failure of the save operation.
-   *
-   * Success result is returned if the checkpoint is saved successfully. The success result
-   * returned has the value
-   * [ExplorationCheckpointDatabaseState.CHECKPOINT_DATABASE_SIZE_LIMIT_EXCEEDED]
-   * if the database has exceeded the size limit of [explorationCheckpointDatabaseSizeLimit],
-   * otherwise the success result is returned with the value
-   * [ExplorationCheckpointDatabaseState.CHECKPOINT_DATABASE_SIZE_LIMIT_NOT_EXCEEDED].
+   * @return a [Deferred] that upon completion indicates the current [CheckpointState]. If the size
+   *     of the checkpoint database is less than the allocated limit of
+   *     [ExplorationStorageDatabaseSize] then the deferred upon completion gives the result
+   *     [CheckpointState.CHECKPOINT_SAVED_DATABASE_NOT_EXCEEDED_LIMIT]. If the size of the
+   *     checkpoint database exceeded [ExplorationStorageDatabaseSize] then
+   *     [CheckpointState.CHECKPOINT_SAVED_DATABASE_EXCEEDED_LIMIT] is returned upon successful
+   *     completion of deferred
```

Comment:
Indentation here seems off. Maybe follow the same pattern as the lines above (i.e. put `If the size...` onwards as a new paragraph

Reply:
done